airflow/www/fab_security/sqla/manager.py

```diff
@@ -580,3 +595,32 @@ def remove_permission_from_role(self, role: Role, permission: Permission) -> Non
             except Exception as e:
                 log.error(c.LOGMSG_ERR_SEC_DEL_PERMROLE.format(str(e)))
                 self.get_session.rollback()
+
+    def remove_permissions_from_role(self, role: Role, permissions: Iterable[Permission]) -> None:
+        """
+        Atomically remove multiple permission pairs from a role in a single transaction.
+
+        :param role: User role containing permissions.
+        :param permissions: Object representing multiple resource->action pairs
+        """
+        role_permissions = set(role.permissions)
+        non_assigned_permissions = []
+        for permission in permissions:
+            if permission not in role_permissions:
+                non_assigned_permissions.append(permission)
+        if non_assigned_permissions:
+            raise ValueError(
+                f"Permissions {non_assigned_permissions} are not assigned to role {role.name}, "
+                "so cannot be removed from it."
+            )
```

Comment:
Probably it still may be prone to race condition and require more isolation, but not sure how far we should go if we go this way :)